File: SWEA/D3/1225. ［S／W 문제해결 기본］ 7일차 － 암호생성기/［S／W 문제해결 기본］ 7일차 － 암호생성기.py
from collections import deque

for _ in range(10):
	tc = int(input())
	q = deque(list(map(int, input().split())))
	idx = 1
	while True:
		num = q.popleft()
		num -= idx
		idx+=1
		if idx==6: idx=1
		if num<=0: q.append(0); break
		q.append(num)
		
	print(f'#{tc} ', end='')
	print(*q)

File: SWEA/D3/1225. ［S／W 문제해결 기본］ 7일차 － 암호생성기/［S／W 문제해결 기본］ 7일차 － 암호생성기.java
import java.io.*;
import java.util.*;

public class Solution {
    static BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
    static StringBuilder sb = new StringBuilder();
    static StringTokenizer st;

    public static void main(String[] args) throws Exception {
        int T = 10;
        while (T-- > 0) {
            int t = Integer.parseInt(input.readLine());
            Deque<Integer> dq = new ArrayDeque<>();
            st = new StringTokenizer(input.readLine());
            for (int i = 0; i < 8; i++) {
                dq.addLast(Integer.parseInt(st.nextToken()));
            }
            int idx=1;
            while (true) {
                int k = dq.pollFirst();
                k -= idx;
                if (k <= 0){
                    k = 0;
                    dq.addLast(k);
                    break;
                }
                dq.addLast(k);
                idx += 1;
                if (idx == 6) idx = 1;
            }

            sb.append("#").append(t).append(" ");
            for (int e : dq) {
                sb.append(e).append(" ");
            }
            sb.append("\n");
        }
        System.out.println(sb.toString());
    }
}